### PR TITLE
Add support for filtering raw events and allowing empty data streams.

### LINKED ISSUE #78: Allow filtering returned raw events
There should be a way to return None from a raw event postprocessing function and have it filtered out of the resulting `DataFrame`.

### PR DESCRIPTION
**Pull request by timburke**
This PR changes how postprocessing of raw events happens in two ways:

1. It passes in the event summary data as well as the event timestamp into the postprocess function so that it has access to this information.
2. It allows the postprocess function to return None, indicating that the event should be filtered out of the resulting set.

Closes #78

### CODE REVIEW

File: iotile_analytics_core/iotile_analytics/core/group.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def find_stream(self, partial, include_empty=False):
 
         return found[0]
 
-    def fetch_stream(self, slug_or_name):
+    def fetch_stream(self, slug_or_name, allow_empty=False):
```
Comment (slack0) [inline iotile_analytics_core/iotile_analytics/core/group.py]: In the case of `allow_empty=True`, does `fetch_stream` what is the return type of the empty stream?

File: iotile_analytics_core/iotile_analytics/core/group.py [hunk 2]
```diff
@@ -203,13 +203,15 @@ def fetch_stream(self, slug_or_name):
                 can be a partial match to a full stream slug or name so long
                 as it uniquely matches.  This is passed to find_stream so anything
                 that find_stream accepts will be accepted here.
+            allow_empty (bool): Allow fetching an empty stream.  Otherwise an ArgumentError
```
Comment (slack0) [inline iotile_analytics_core/iotile_analytics/core/group.py]: Checking the wording here. When will an `ArgumentError` be raised? When the target stream is empty, while `allow_empty` is `True`?

File: iotile_analytics_core/iotile_analytics/core/group.py (hunk 1 above)
Comment (timburke) [reply iotile_analytics_core/iotile_analytics/core/group.py]: It should be an empty `StreamSeries`, test coverage is here:
https://github.com/iotile/iotile_analytics/pull/79/files#diff-be9d8aa29725634a5bce735649896aa0R99

File: iotile_analytics_core/iotile_analytics/core/group.py (hunk 2 above)
Comment (timburke) [reply iotile_analytics_core/iotile_analytics/core/group.py]: Updated the wording:

```
allow_empty (bool): Allow fetching an empty stream.  If allow_empty is False or
       not passed, an ArgumentError will be raised if the target stream is empty.
```

**Pull request merged**